response_play.go

```diff
@@ -0,0 +1,111 @@
+package pagerduty
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+type ResponsePlay struct {
+	APIObject
+	Name        string        `json:"name,omitempty"`
+	Description string        `json:"description,omitempty"`
+	Team        *APIReference `json:"team,omitempty"`
+}
+
+type ListResponsePlaysResponse struct {
+	APIListObject
+	ResponsePlays []ResponsePlay `json:"response_plays"`
+}
+
+// ListResponsePlaysWithContext lists existing response plays.
+func (c *Client) ListResponsePlaysWithContext(ctx context.Context) (*ListResponsePlaysResponse, error) {
+	resp, err := c.get(ctx, "/response_plays")
+	if err != nil {
+		return nil, err
+	}
+
+	var result ListResponsePlaysResponse
+	if err = c.decodeJSON(resp, &result); err != nil {
+		return nil, err
+	}
+
+	return &result, nil
+}
+
+// CreateResponsePlayWithContext creates a new response play.
+func (c *Client) CreateResponsePlayWithContext(ctx context.Context, rp ResponsePlay) (*ResponsePlay, error) {
+	d := map[string]ResponsePlay{
+		"response_play": rp,
+	}
+
+	resp, err := c.post(ctx, "/response_plays", d, nil)
+	return getResponsePlayFromResponse(c, resp, err)
+}
+
+// GetResponsePlayWithContext gets details about an existing response play.
+func (c *Client) GetResponsePlayWithContext(ctx context.Context, id string) (*ResponsePlay, error) {
+	resp, err := c.get(ctx, "/response_plays/"+id)
+	return getResponsePlayFromResponse(c, resp, err)
+}
+
+// UpdateResponsePlayWithContext updates an existing response play.
+func (c *Client) UpdateResponsePlayWithContext(ctx context.Context, rp ResponsePlay) (*ResponsePlay, error) {
+	d := map[string]ResponsePlay{
+		"response_play": rp,
+	}
+
+	resp, err := c.put(ctx, "/response_plays/"+rp.ID, d, nil)
+	return getResponsePlayFromResponse(c, resp, err)
+}
+
+// DeleteResponsePlayWithContext deletes an existing response play.
+func (c *Client) DeleteResponsePlayWithContext(ctx context.Context, id string) error {
+	_, err := c.delete(ctx, "/response_plays/"+id)
+	return err
+}
+
+// RunResponsePlayWithContext runs a response play on a given incident.
+func (c *Client) RunResponsePlayWithContext(ctx context.Context, rp ResponsePlay, incident Incident, from string) error {
```

Comment:
I think this should take the response play ID as an argument, and not an entire `ResponsePlay`.